Use closest previous version when looking for stable Scala versions

alexarchambault opened the pull request:
This should allow not to have to update https://github.com/VirtusLab/scala-cli-scala-versions/blob/main/scala-versions-v1.json upon each release. This file only needs to be updated when new Scala versions are released (and known to be supported by Scala CLI).

lwronski reviewed (commented):
For me, only the unit test is missing, the rest LGTM

a unit test might look like this:
```
val confFile = s"""[
                    | {
                    |  "scalaCliVersion": "0.1.0",
                    |  "supportedScalaVersions": ["3.0.1", "2.13.4", "2.12.13"]
                    | }
                    |]""".stripMargin

 test(s"Use closest previous version when looking for stable Scala versions") {
    TestInputs.withTmpDir("conf-scala-versions") { dirPath =>

      val confFilePath = dirPath / "conf-file.json"
      os.write(confFilePath, confFile)

      val options = BuildOptions(
        scalaOptions = ScalaOptions(
          scalaVersion = Some(MaybeScalaVersion("3")),
          scalaBinaryVersion = None,
          supportedScalaVersionsUrl = Some(s"file://${confFile2.toString()}")
        )
      )

      val scalaParams          = options.scalaParams.orThrow.getOrElse(???)
      val expectedScalaVersion = "3.0.1"
      val expectedScalaParams  = ScalaParameters(expectedScalaVersion)

      expect(scalaParams == expectedScalaParams)
    }
  }
```

alexarchambault committed:
Small refacto in BuildOptionsTests

- remove unnecessary None values
- more BuildOptions refacto
- better os.Path to URL string conversion in BuildOptionsTests

alexarchambault committed:
Add extra data in supported Scala version test

And use a custom Scala CLI version in the tests

alexarchambault committed:
Use closest previous version when looking for stable Scala versions
